lib/interface_quda.cpp

```diff
@@ -4286,6 +4347,262 @@ void momResidentQuda(void *mom, QudaGaugeParam *param)
     delete momResident;
     momResident = nullptr;
     profileGaugeForce.TPSTOP(QUDA_PROFILE_FREE);
+
+    // if we used FGI, free memory for momResidentCopy
+    if (momResidentCopy) {
+      delete momResidentCopy;
+      momResidentCopy = nullptr;
+    }
+    profileGaugeForce.TPSTOP(QUDA_PROFILE_FREE);
+  }
+
+  profileGaugeForce.TPSTOP(QUDA_PROFILE_TOTAL);
+}
+
+void copyGaugeQuda()
+{ // FGI make a copy of the precise gauge field on device
+  profileGaugeForce.TPSTART(QUDA_PROFILE_TOTAL);
+
+  // Wilson Links
+  //----------------------------------------------------------
+  if(gaugePrecise) {
```

Comment:
Just confirming --- it's intentional that this doesn't throw an `errorQuda` if `gaugePrecise` is unset, right? I see that it's fully safe code on its own.